app/src/main/java/com/ldlda/chesscom_stats/LessonsFragment.java

```diff
@@ -8,12 +8,53 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.ldlda.chesscom_stats.adapter.LessonAdapter;
+
+import java.util.Arrays;
+import java.util.List;
 
 public class LessonsFragment extends Fragment {
-    //Inflate fragment
+
+    private RecyclerView recyclerView;
+
+    public LessonsFragment() {
+        // Required empty public constructor
+    }
+
     @Nullable
     @Override
-    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
-        return inflater.inflate(R.layout.fragment_lessons, container, false);
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_lessons, container, false);
+
+        recyclerView = view.findViewById(R.id.recyclerView);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+
+        List<String> lessonTitles = Arrays.asList(
+                "Lesson 1:",
+                "Lesson 2:",
+                "Lesson 3:",
+                "Lesson 4:",
+                "Lesson 5:"
+        );
+
+        LessonAdapter adapter = new LessonAdapter(lessonTitles, this::openLesson);
+        recyclerView.setAdapter(adapter);
+
+        return view;
+    }
+
+    private void openLesson(int lessonIndex) {
+        LessonContents contentLesson = LessonContents.newInstance(lessonIndex);
+
+        FragmentManager fragmentManager = requireActivity().getSupportFragmentManager();
+        fragmentManager.beginTransaction()
+                .replace(R.id.fragment_container, contentLesson, "ContentLesson")
+                .addToBackStack(null)
+                .commit();
```

Comment:
fuck idek know

the furthest ive been is to replace replace with hide(this).add. 

This system is very brittle.

```suggestion
        fragmentManager.beginTransaction()
                .hide(this)
                .add(R.id.fragment_container, contentLesson, "ContentLesson")
                .addToBackStack("gay")
                .commit();
```

Reply:
at MainActivity.java:
```diff
@@ -55,6 +56,8 @@ public class MainActivity extends AppCompatActivity {
             int menuId = item.getItemId();
             Fragment target = fragmentFor(menuId);
             if (target != null && target != currentFragment) {
+                getSupportFragmentManager()
+                        .popBackStack("gay", FragmentManager.POP_BACK_STACK_INCLUSIVE);
                 getSupportFragmentManager()
                         .beginTransaction()
                         .hide(currentFragment)

```